azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -1776,12 +1776,49 @@ private void processFlow(ExecutionReference reference,
       }
     }
 
+    /* Helper method to fetch  overriding Executor, if a valid user has specifed otherwise return null */
```

Comment:
I guess my question is, 
might the caller function/code want to know which exception scenario it ran into?

If you have it in logs humans will understand why but it would be manual intervention